### DISCUSSION TITLE
auth.email() function not exist in row-level security policy

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#1147] [Questions] [answered]
# Bug report

## Describe the bug

When I set a row-level security policy using the function auth.email(), it says:
function auth.email() does not exist

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. create a table
2. set a row-level security policy for that table
3. set the definition to be "right(auth.email(), 13) = '@blizzard.com'" for example
4. click 'save'
5. see error: function auth.email() does not exist

## Expected behavior

no error, policy saved successfully

## Screenshots

![image](https://user-images.githubusercontent.com/49301824/113321280-053fe780-9346-11eb-91c2-616393447157.png)
![image](https://user-images.githubusercontent.com/49301824/113321380-1c7ed500-9346-11eb-93cd-44fd1faa0b53.png)

## System information

- OS: Windows 10
- Browser: chrome

**Comment by steve-chavez** [comment, marked as answer]
For now you can create the function yourself:

```sql
create or replace function auth.email() returns text as $$
  select nullif(current_setting('request.jwt.claim.email', true), '')::text;
$$ language sql;
```

**Reply by steve-chavez** [reply]
New projects include this function by default.

**Comment by kiwicopple** [comment]
If anyone else is experiencing this, it's likely because you're on a very old database. We didn't run this as a migration across all of the projects. 

I will transfer this over to Discussions so that it's easy to find in the future

**Reply by easy-street** [reply]
to confirm, it's expected that the `auth.email` function should exist in new projects? seems to be missing in new projects currently.

**Reply by HosAkh** [reply]
Yea my project and account are a week old and it's missing it.

**Reply by heauton** [reply]
Hmm. Same here.

**Reply by inian** [reply]
Hi everyone, I can confirm that we have not added this to new projects, so you will need to define this function after the project launches.

**Reply by inian** [reply]
Since there has been some demand for this, new projects launched as of this minute will contain the `auth.email` function by default.
